clients/java/src/main/java/io/zeebe/client/impl/ZeebeCallCredentials.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.zeebe.client.impl;
+
+import io.grpc.Metadata;
+import io.grpc.SecurityLevel;
+import io.zeebe.client.CredentialsProvider;
+import java.util.concurrent.Executor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZeebeCallCredentials extends io.grpc.CallCredentials {
+  private static final Logger LOG = LoggerFactory.getLogger("io.zeebe.client.impl");
+
+  private final CredentialsProvider credentialsProvider;
+
+  ZeebeCallCredentials(final CredentialsProvider credentialsProvider) {
+    this.credentialsProvider = credentialsProvider;
+  }
+
+  @Override
+  public void applyRequestMetadata(
+      final RequestInfo requestInfo, final Executor appExecutor, final MetadataApplier applier) {
+
+    if (requestInfo.getSecurityLevel().ordinal() < SecurityLevel.PRIVACY_AND_INTEGRITY.ordinal()) {
+      LOG.warn(
+          "The request's security level does not guarantee that the credentials will be confidential.");
+    }
+
+    final Metadata headers = new Metadata();
+    credentialsProvider.applyCredentials(headers);
+    applier.apply(headers);
+  }
+
+  @Override
+  public void thisUsesUnstableApi() {}
```

Comment:
Are we supposed to implement something here? :thinking:

Reply:
No, it's just to make even more explicit that the API is unstable. Here's the docs https://grpc.github.io/grpc-java/javadoc/io/grpc/CallCredentials.html#thisUsesUnstableApi--